lips/lip-10.md

```diff
@@ -0,0 +1,234 @@
+* Only special Treasury Compliance account(address `0xB1E55D`) can manipulate LibraIdDomains resource:
+  * Only TC account can create and publish LibraIdDomains resource
+  * Only TC account can add, remove or update an LibraIdDomain within LibraIdDomains resource
+* In order to register a Libra Id domain, VASP needs to submit a request to Libra Association. LA will perform certain checks(out of scope of this document) before they will issue an on-chain transaction to register a Libra Id Domain. The checks that LA will perform will make sure that VASPs do not claim irrelevant Libra Id Domains. LA submission will also ensure that Libra Id Domains are unique.
+* It should also be possible(and probably going to be most common) for VASPs to submit their desired Libra ID domain(s) during the registration.
+* Association may provide an open source application(*Indexer*) that will allow indexing available Libra IDs and will provide a convenient REST api for applications to fetch information about the domains. The API of such an indexer is out of scope of this RFC.
+
+# Peer-2-peer payment protocol
+One of the main use cases for Libra ID is transferring money from one user to another across different wallets.
+
+<img alt="Payment flow" src={useBaseUrl('img/libra-id-flow.png')} />
+
+## UI flow
+* Receiver shares their Libra ID with sender
+* Sender enters receiver’s Libra ID(and other txn details such as amount) in their wallet app
+* Sender’s wallet can fetch profile picture and full name of the receiver prior to sending money, and show it to sender, to improve security (if receiver VASP allows it)
+* Sender presses ‘send’ button
+* Receiver receives money and information about the sender (optionally)
+
+## Backend flow
+In order to support peer-2-peer payments, VASPs need to implement two Libra ID endpoints:
+
+
+* **Info endpoint** for querying optional information about the receiver. VASPs can provide this endpoint to improve UX and security of payment, but they are not obligated to - VASPs may choose not to share any information about the user via this endpoint, including whether the user exists or not
+* **Payment endpoint** to perform actual money transfer.
+
+Common considerations:
+
+* For both endpoints, all communications are signed via the same mechanism as for other off-chain communications, see the [Payloads section of lip-1](https://github.com/libra/lip/blob/master/lips/lip-1.mdx#payloads) for details. As such, all vasp-2-vasp communications are authenticated.
+
+* All Libra ID endpoints are nested under `pay.<version>` URI path:
+
+     `https://<hostname>:<port>/pay.<protocol_version>/<endpoint>`
+
+* All Libra ID endpoints use the same `base_url` and same `compliance_public_key` for authentication as other off-chain APIs.
```

Comment:
Does using `compliance_public_key` make sense here? We need it for for TR because the signature will be checked on-chain, but LibraID seems like it could use a more conventional authentication mechanism. Not my area of expertise at all, but maybe someone who knows a lot about web authentication could weigh in.

Reply:
Yeah, I think currently `compliance_public_key` is simply overloaded - it is used to both signing KYC metadata, _and_ for signing the requests. However, this is how things are set in lip-1(main offchain lip), and this document just re-states it for clarity

Reply:
I don't follow, that compliance key is an ed25519 key for signing metadata that appears on chain. I don't know why we need a compliance key in this place that is strictly used for dual attestation. This is too overloaded and has no meaning in the rest of this document. Also if we add an IP for each address, what are we doing? Can we clarify the purpose of a wallet uri?